### PR TITLE
TIG-2082: Simple Actor that periodically logs a message

### PR DESCRIPTION
**Pull request by rtimmons**

### CODE REVIEW

File: src/cast_core/src/LoggingActor.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    explicit PhaseConfig(PhaseContext& phaseContext)
+        : logEvery{phaseContext["LogEvery"].to<TimeSpec>()}, started{metrics::clock::now()} {
+        if (phaseContext["Blocking"].to<std::string>() != "None") {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("LoggingActor must have Blocking:None"));
+        }
+    }
+    void report() {
+        // Avoid calling now() on most iterations
+        if (++iteration % 10000 != 0) {
```
Comment (guoyr) [inline src/cast_core/src/LoggingActor.cpp]: It should be possible to add a `__builtin_expect` compiler hint here if you want to be extra-fancy, but probs not worth it.

File: src/cast_core/test/LoggingActor_test.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+)",
+                          ""};
+        ActorHelper ah(config.root(), 2, MongoTestFixture::connectionUri().to_string());
+        ah.run();
+        // Don't actually assert anything because it's hard to assert interactions with logging.
+        // If you're running this test manually you should see exactly 3 log messages from
+        // the LoggingActor.
+    }
+}
+}  // namespace
```
Comment (guoyr) [inline src/cast_core/test/LoggingActor_test.cpp]: 💅

File: src/workloads/docs/LoggingActor.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+SchemaVersion: 2018-07-01
```
Comment (guoyr) [inline src/workloads/docs/LoggingActor.yml]: `LoggingActorExample.yml`? So people know it's not meant to be run as a standalone workload?

File: src/cast_core/src/LoggingActor.cpp (hunk 1 above)
Comment (rtimmons) [reply src/cast_core/src/LoggingActor.cpp]: Ya so far we've stayed pretty much entirely on non-vendor-specific C++17 so I'm hesitant to venture into extensions for something as simple as this.

**Pull request merged**